spec/services/punches_service_spec.rb

```diff
@@ -0,0 +1,82 @@
+require 'rails_helper'
+
+RSpec.describe PunchesService do
+  let(:company) { create(:company) }
+  let(:user) { create(:user, :with_token, company: company) }
+  let(:project) { create(:project, company: user.company) }
+
+  describe "#create" do
+    context "with right arguments" do
+      let(:punches) {
+        [
+          {
+            "from" => "2022-04-19T12:00:00",
+            "to" => "2022-04-19T15:00:00",
+            "project_id" => "#{project.id}"
+          },
+          {
+            "from" => "2022-04-19T16:00:00",
+            "to" => "2022-04-19T21:00:00",
+            "project_id" => "#{project.id}"
+          }
+        ]
+      }
+
+      let(:office) { create(:office, :with_holiday) }
+
+      it "returns the created punches" do
+        created_punches = described_class.create(punches, user).map { |punch| {from: punch['from'].strftime("%FT%T"), to: punch['to'].strftime("%FT%T"), project_id: punch['project_id'].to_s } }
+        expected_punches = punches.map { |punch| {from: punch["from"], to: punch["to"], project_id: punch["project_id"].to_s } }
+        expect(created_punches).to match_array(expected_punches)
+      end
+    end
+
+    context 'when does have another punch on same date' do
```

Comment:
I believe by using a bit more FactoryBot you'd make your spec more simple.
```ruby
context 'when does have another punch on same date' do
  let!(:punch_1) { create(:punch, from: '2022-04-19T12:00:00', to: '2022-04-19T15:00:00', project: project) }
  let!(:punch_2) { create(:punch, from: '2022-04-19T16:00:00', to: '2022-04-19T21:00:00', project: project) }

  let(:punches) do
    [
      {
        "from" => "2022-04-19T10:00:00",
        "to" => "2022-04-19T12:00:00",
        "project_id" => project.id
      },
      {
        "from" => "2022-04-19T15:00:00",
        "to" => "2022-04-19T20:00:00",
        "project_id" => project.id
      }
    ]
  end

  it 'returns created punches and deletes the last punches' do
    result = described_class.create(punches, user)

    expect(Punch.count).to eq(2)
    expect(result).to match_array(punches)
  end
end
 ```

Reply:
Yes, It makes more simple, I will refactor this